compose/cli/main.py

```diff
@@ -1298,31 +1298,25 @@ def build_one_off_container_options(options, detach, command):
 
 def run_one_off_container(container_options, project, service, options, toplevel_options,
                           toplevel_environment):
-    if not options['--no-deps']:
-        deps = service.get_dependency_names()
-        if deps:
-            project.up(
-                service_names=deps,
-                start_deps=True,
-                strategy=ConvergenceStrategy.never,
-                rescale=False
-            )
-
-    project.initialize()
-
-    container = service.create_container(
-        quiet=True,
+    detach = options.get('--detach')
+    use_network_aliases = options.get('--use-aliases')
+    containers = project.up(
+        service_names=[service.name],
+        start_deps=not options['--no-deps'],
+        strategy=ConvergenceStrategy.never,
+        detached=detach,
+        rescale=False,
         one_off=True,
-        **container_options)
-
-    use_network_aliases = options['--use-aliases']
+        override_options=container_options,
+    )
+    container = [c for c in containers if c.service == service.name][0]
```

Comment:
Could we do a check before accessing this please? Just to make sure we won't throw an exception to users.

Reply:
Of course 👍 I've modified this to raise an `OperationFailedError` in case target service isn't in the container list. This is what it'll look like to a user in case they somehow manage to get in this state:
```
$ docker-compose run --rm test sh
Creating docker-compose-example_db_1 ... done
Creating docker-compose-example_test_run ... done
ERROR: Could not bring up the requested service
```